src/libsyncengine/update_detection/file_system_observer/folderwatcher_mac.cpp

```diff
@@ -40,17 +40,17 @@ static void callback([[maybe_unused]] ConstFSEventStreamRef streamRef, void *cli
     }
 
     static const FSEventStreamEventFlags interestingFlags =
-        kFSEventStreamEventFlagItemCreated         // for new folder/file
-        | kFSEventStreamEventFlagItemRemoved       // for rm
-        | kFSEventStreamEventFlagItemInodeMetaMod  // for mtime change
-        | kFSEventStreamEventFlagItemRenamed       // also coming for moves to trash in finder
-        | kFSEventStreamEventFlagItemModified      // for content change
-        | kFSEventStreamEventFlagItemChangeOwner;  // for rights change
+            kFSEventStreamEventFlagItemCreated // for new folder/file
+            | kFSEventStreamEventFlagItemRemoved // for rm
+            | kFSEventStreamEventFlagItemInodeMetaMod // for mtime change
+            | kFSEventStreamEventFlagItemRenamed // also coming for moves to trash in finder
+            | kFSEventStreamEventFlagItemModified // for content change
+            | kFSEventStreamEventFlagItemChangeOwner; // for rights change
 
     std::list<std::pair<std::filesystem::path, OperationType>> paths;
-    CFArrayRef eventPaths = (CFArrayRef)eventPathsVoid;
+    CFArrayRef eventPaths = (CFArrayRef) eventPathsVoid;
     for (int i = 0; i < static_cast<int>(numEvents); ++i) {
-        OperationType opType;
+        OperationType opType = OperationType::None;
```

Comment:
**Suggestion.** Make a small function that computes `opType` and write units test for it.

Reply:
Done in commit f5bb0a5